GTG/gtk/browser/main_window.py

```diff
@@ -1285,9 +1285,12 @@ def on_dismiss_task(self, widget=None):
     def apply_filter_on_panes(self, filter_name, refresh=True, parameters=None):
         """ Apply filters for every pane: active tasks, closed tasks """
 
+        current_pane = self.get_selected_pane()
         for pane in self.vtree_panes:
             vtree = self.req.get_tasks_tree(name=pane, refresh=False)
-            vtree.apply_filter(filter_name, refresh=refresh, parameters=parameters)
+            vtree.apply_filter(filter_name,
+                               refresh=current_pane == pane,
```

Comment:
Given how important this seems to be to maintain somewhat decent performance, I think it might be good to add some comment(s) in the code to explain the gotchas / warn people to only refresh the currently shown pane to cut down the performance impact massively, avoiding potential future naïve code mistakes?